src/main/java/org/folio/edgecommonspring/security/SecurityManagerService.java

```diff
@@ -0,0 +1,126 @@
+package org.folio.edgecommonspring.security;
+
+import static java.util.Optional.ofNullable;
+import static org.folio.edge.api.utils.Constants.DEFAULT_SECURE_STORE_TYPE;
+import static org.folio.edge.api.utils.Constants.PROP_SECURE_STORE_TYPE;
+import static org.folio.edge.api.utils.Constants.X_OKAPI_TOKEN;
+
+import java.util.Properties;
+import javax.annotation.PostConstruct;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.lang3.StringUtils;
+import org.folio.edge.api.utils.cache.TokenCache;
+import org.folio.edge.api.utils.cache.TokenCache.NotInitializedException;
+import org.folio.edge.api.utils.model.ClientInfo;
+import org.folio.edge.api.utils.security.SecureStore;
+import org.folio.edge.api.utils.security.SecureStore.NotFoundException;
+import org.folio.edge.api.utils.security.SecureStoreFactory;
+import org.folio.edgecommonspring.client.AuthnClient;
+import org.folio.edgecommonspring.domain.entity.ConnectionSystemParameters;
+import org.folio.edgecommonspring.exception.AuthorizationException;
+import org.folio.edgecommonspring.util.ApiKeyUtils;
+import org.folio.edgecommonspring.util.PropertiesUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class SecurityManagerService {
+
+  @Autowired
+  private final AuthnClient authnClient;
+  private SecureStore secureStore;
+  private TokenCache tokenCache;
```

Comment:
Maybe we can use here production-ready cache, for example, Hazelcast, instead of TokenCache. @craigmcnally , it will be useful to hear your opinion on it.

Reply:
As for me, it's good idea to try it.
Here example how we can use it: https://dzone.com/articles/caching-with-spring-boot-and-hazelcast

Reply:
As far as I understood, we should use more existing logic from edge-api-utils, that`s why I used this cache

Reply:
Initially, let's use what we have in edge-api-utils.  We don't want to increase the scope of this right now.  Going forward, we can look at replacing that with something else.  One thing to keep in mind is that ideally the behavior of edge APIs developed using edge-common-spring is very close if not identical to edge APIs developed using edge-common.  That includes the cache behavior.  This cache is extremely simple, and doesn't attempt to be cluster aware - there's no need IMO for that complexity in the context of a token cache.  

Also note, that we'll likely need to do some work in this area in R2/R3 given that the core platform team has picked up the effort to expire okapi-tokens after a relatively short period of time.  So our edge-common frameworks will need to adjust the token caches and will need to incorporate usage of refresh tokens, etc.  This may be a good opportunity to revisit the cache situation.